File: 0002/problem2.py
def even(x):
	return x % 2 == 0

def problem2(n):
	sum = 0
	a, b = 0, 1
	while a < n:
		print(a, end=' ')
		a, b = b, a+b
		
		if even(a): sum += a

	print()
	print("sum: {}".format(sum))
	
	
problem2(4000000)

File: 0002/problem2.swift
func even(x: Int) -> Bool {
    return x % 2 == 0
}

func problem2(max: Int) {
    var sum = 0
    var a = 0
    var b = 1
    while b < max {
        (a,b) = (b, a+b)
        
        if even(x: a){
            sum += a
        }
    }
    
    print("sum: \(sum)")
}

problem2(max: 4000000)
